src/validations/sch/sch.sch

```diff
@@ -159,13 +175,139 @@
+            <sch:let
+                name="assertions"
+                value="//(sch:assert | sch:report)" />
+            <sch:let
+                name="assertion-ids"
+                value="$assertions/@id" />
+            <sch:let
+                name="tests"
+                value="$xspec//(x:expect-not-assert | x:expect-assert)[not(@pending)]" />
+            <sch:let
+                name="referenced-tests"
+                value="$tests[@id = $assertion-ids]" />
+            <sch:let
+                name="coverage"
+                value="count($referenced-tests) div (count($assertions) * 2)" />
+            <sch:report
+                id="report-test-coverage"
+                role="information"
+                diagnostics="report-test-coverage-diagnostic"
+                test="true()">There are <sch:value-of
+                    select="count($assertions)" /> assertions and there are <sch:value-of
+                    select="count($referenced-tests)" /> unit tests which reference those assertions for test coverage of <sch:value-of
+                    select="format-number($coverage, '09.99%')" />.</sch:report>
+
+        </sch:rule>
+
+    </sch:pattern>
+
+    <sch:pattern
+        id="FedRAMP-extensions">
+
+        <sch:rule
+            context="sch:assert">
+
+            <sch:assert
+                sqf:fix="insert-doc-reference-attribute"
+                test="@doc:reference"
+                role="warning"
+                id="has-doc-reference">Every rule has a doc:reference attribute.</sch:assert>
+
+            <sqf:fix
+                id="insert-doc-reference-attribute">
+                <sqf:description>
+                    <sqf:title>Add a doc:reference attribute</sqf:title>
+                </sqf:description>
+                <sqf:user-entry
+                    name="URI">
+                    <sqf:description>
+                        <sqf:title>Enter a URI</sqf:title>
+                    </sqf:description>
+                </sqf:user-entry>
+                <sqf:add
+                    match="."
+                    node-type="attribute"
+                    target="doc:reference"
+                    select="'reference'" />
+            </sqf:fix>
+
+        </sch:rule>
+
     </sch:pattern>
 
     <sch:diagnostics>
         <sch:diagnostic
             id="xspec-defined-and-available-diagnostic">@href present and available.</sch:diagnostic>
         <sch:diagnostic
-            id="xspec-resolved-uri-diagnostic">@href resolves.</sch:diagnostic>
+            id="xspec-resolved-uri-diagnostic">This is informational message (not an error).</sch:diagnostic>
+        <sch:diagnostic
+            id="report-test-coverage-diagnostic">This is informational message (not an error).</sch:diagnostic>
+        <sch:diagnostic
+            id="context-reuse-diagnostic">This sch:rule context is used elsewhere. This can cause XSpec evaluation problems.</sch:diagnostic>
```

Comment:
B: Can we discuss in Slack what kind of problems to be more specific or link to a reference explanation?It would probably help me to know or understand this better so we all know how to avoid it!